sklearn/cross_decomposition/_pls.py

```diff
@@ -85,7 +85,7 @@ def _get_first_singular_vectors_svd(X, Y):
     Here the whole SVD is computed.
     """
     C = np.dot(X.T, Y)
-    U, _, Vt = svd(C, full_matrices=False)
+    U, _, Vt = svd(C, full_matrices=False, check_finite=False)
```

Comment:
Because we divide by X std on line 108, I think this may be non finite if the std is small enough (same reason as https://github.com/scikit-learn/scikit-learn/pull/18850#discussion_r526344155)

Reply:
Also the dot product may return infinite values (same as https://github.com/scikit-learn/scikit-learn/pull/18848/files#r531103450 )

Reply:
@mathschy thank you for your review, yes i think  you're right about the dot product so i'll revert the commit in line 88 but i will wait for another review to talk about the division by X_std . again thank you